pyproject.toml

```diff
@@ -106,13 +106,19 @@ sdp = [
 ]
 test = [
     "ci-watson>=0.5.0",
-    "ruff",
     "pysiaf>=0.13.0",
     "pytest>=6.0.0",
     "pytest-cov>=2.9.0",
     "pytest-doctestplus>=0.10.0",
     "requests_mock>=1.0",
 ]
+dev = [
```

Comment:
If we go this route (adding version specifications in in pyproject.toml and pre-commit-config.yaml) can we name this something like `lint` instead of `dev`?

Reply:
I think it would be clean to have the dev be the only tag necessary to get all necessary packages for development. We could include the new dependencies in a `lint` group, then have `dev` include `test` and `lint`. Or does that not address what you're after?

Reply:
The more I think about this the more I leaving these tools up to `pre-commit` to install and manage versions is the easiest and most robust solution.
re: https://github.com/spacetelescope/jwst/pull/9097#issuecomment-2619286644